lib/templates/src/App.vue

```diff
@@ -11,18 +11,70 @@
       </Screen>
     </template>
 
-    <template #[screen]="{ trial }" v-for="screen in 2">
-      <AudioDiscriminationWithPriming
-        :priming-audio="trial.audio.primingAudio"
-        :trial-audio="trial.audio.trialAudio"
-        :option1="trial.options.option1"
-        :option2="trial.options.option2"
-        :key="screen"
-      >
-      </AudioDiscriminationWithPriming>
+    <template #1="{ nextScreen, addResult, trial }">
+      <Screen>
+        <ForcedChoiceInput
+          question="How did you like this?"
+          :options="[trial.options.option1, trial.options.option2]"
+          :answer.sync="answer"
+          @change:answer="
+            addResult({ answer, options: trial.options });
+            nextScreen();
+          "
+        />
+      </Screen>
+    </template>
+
+    <template #2="{ nextScreen, addResult, trial }">
+      <Screen>
+        <ImageSelectionInput
+          question="How did you like this?"
+          :options="[
+            { label: trial.options.option1, src: 'img/1.jpg' },
+            { label: trial.options.option2, src: 'img/2.jpg' }
+          ]"
+          :answer.sync="answer"
+          @change:answer="
+            addResult({ answer, options: trial.options });
+            nextScreen();
+          "
+        />
+      </Screen>
+    </template>
+
+    <template #3="{ nextScreen, addResult, trial }">
+      <Screen>
+        <TextareaInput
+          question="How did you like this?"
+          :answer.sync="answer"
+          @change:answer="
+            addResult({ answer });
+            nextScreen();
+          "
+        />
+      </Screen>
+    </template>
+
+    <template #4="{ nextScreen, addResult, trial }">
+      <Screen>
+        <SliderInput
+          question="How did you like this?"
+          left="bad"
+          right="good"
+          :answer.sync="answer"
+        />
+        <button
+          @click="
+            addResult({ answer });
+            nextScreen();
+          "
+        >
+          Next
+        </button>
+      </Screen>
     </template>
 
-    <template #3>
+    <template #5>
```

Comment:
I'm not entirely happy with this solution. Whenever the user wants to change the order they have to change the slot ids as well :/